lib/model/internal_link.dart

```diff
@@ -71,11 +71,14 @@ Uri narrowLink(PerAccountStore store, Narrow narrow, {int? nearMessageId}) {
     fragment.write('${element.operator}/');
 
     switch (element) {
+      case ApiNarrowChannelModern():
       case ApiNarrowStream():
```

Comment:
nit: since these are handled the same way here, they can continue to use the base class with its simpler name:
```suggestion
      case ApiNarrowChannel():
```

(There still shouldn't be an instance of directly the base class here, because the `operator` getter above would complain. But this code doesn't need to go out of its way to detect that.)

This differs from the ApiNarrowDm logic below, because for that one the operand looks different between the two cases.